src/lib/y2network/sysconfig_routes_file.rb

```diff
@@ -0,0 +1,194 @@
+  #
+  # @example Reading routes for the interface eth1
+  #   file = Y2Network::SysconfigRoutesFile.new("/etc/sysconfig/network/ifroute-eth1")
+  #   file.load
+  #   file.routes.size #=> 1
+  class SysconfigRoutesFile
+    DEFAULT_ROUTES_FILE = "/etc/sysconfig/network/routes".freeze
+
+    # @return [Array<Route>] Routes
+    attr_reader :routes
+
+    # @param path [String] File path
+    def initialize(file_path = DEFAULT_ROUTES_FILE)
+      @path =
+        if file_path == DEFAULT_ROUTES_FILE
+          Yast::Path.new(".routes")
+        else
+          register_ifroute_agent_for_path(file_path)
+        end
+    end
+
+    # Loads routes from system
+    #
+    # @return [Array<Hash<String, String>>] list of hashes representing routes
+    #                                       as provided by SCR agent.
+    #                                       keys: destination, gateway, netmask, [device, [extrapara]]
+    def load
+      entries = Yast::SCR.Read(path) || []
+      entries = normalize_entries(entries.uniq)
+      @routes = entries.map { |r| build_route(r) }
+    end
+
+  private
+
+    # @return [Yast::Path]
+    attr_reader :path
+
+    MISSING_VALUE = "-".freeze
+    private_constant :MISSING_VALUE
+    DEFAULT_DEST = "default".freeze
+    private_constant :DEFAULT_DEST
+
+    # Converts routes config as read from system into well-defined format
+    #
+    # Expects list of hashes as param. Hash should contain keys "destination",
+    # "gateway", "netmask", "device", "extrapara"
+    #
+    # Currently it converts "destination" in CIDR format (<ip>/<prefix_len>)
+    # and keeps just <ip> part in "destination" and puts "/<prefix_len>" into
+    # "netmask"
+    #
+    # @param routes [Array<Hash>] in quad or CIDR flavors (see {#Routes})
+    # @return [Array<Hash>] in quad or slash flavor
+    def normalize_entries(entries)
+      return entries if entries.nil? || entries.empty?
+
+      entries.map do |entry|
+        subnet, prefix = entry["destination"].split("/")
+
+        next entry if prefix.nil?
+
+        entry["destination"] = subnet
+        entry["netmask"] = "/#{prefix}"
+
+        entry
+      end
+    end
+
+    # Given an IP and a netmask, returns a valid IPAddr object
+    #
+    # @param ip_str      [String] IP address; {MISSING_VALUE} means that the IP is not defined
+    # @param netmask_str [String] Netmask; {MISSING_VALUE} means than no netmask was specified
+    # @return [IPAddr,nil] The IP address or `nil` if the IP is missing
+    def build_ip(ip_str, netmask_str = MISSING_VALUE)
+      return nil if ip_str == MISSING_VALUE
+
+      ip = IPAddr.new(ip_str)
+      netmask_str == MISSING_VALUE ? ip : ip.mask(netmask_str)
+    end
+
+    # Build a route given a hash from the SCR agent
+    #
+    # @param hash [Hash] Hash from the `.routes` SCR agent
+    # @return Route
+    def build_route(hash)
+      # TODO: check whether the iface is configured in the system
+      iface = Interface.new(hash["device"])
```

Comment:
TODO: It loads absence of device as an Interface with "-" name, so we should fix that initialization. When reading from routes it should be :any while when loading from `ifroute` it should be the interface name inferred from the file name